File: reactivando-frontend/src/services/candidatoService.js
import api from './api';

export const candidatoService = {
  obtenerPorId: async (id) => {
    const response = await api.get(`/candidatos/${id}`);
    return response.data;
  },

  obtenerPorUsuarioId: async (usuarioId) => {
    const response = await api.get(`/candidatos/usuario/${usuarioId}`);
    return response.data;
  },

  obtenerTodos: async () => {
    const response = await api.get('/candidatos');
    return response.data;
  },

  actualizarPerfil: async (usuarioId, candidatoData) => {
    const response = await api.put(`/candidatos/usuario/${usuarioId}`, candidatoData);
    return response.data;
  },
};

File: reactivando-backend/src/main/java/com/reactivando/futuro/service/CandidatoService.java
package com.reactivando.futuro.service;

import com.reactivando.futuro.dto.CandidatoRequestDTO;
import com.reactivando.futuro.dto.CandidatoResponseDTO;

import java.util.List;

public interface CandidatoService {
    CandidatoResponseDTO obtenerPorId(Long id);
    CandidatoResponseDTO obtenerPorUsuarioId(Long usuarioId);
    CandidatoResponseDTO obtenerPorCorreo(String correo);
    List<CandidatoResponseDTO> obtenerTodos();
    CandidatoResponseDTO actualizarPerfil(Long usuarioId, CandidatoRequestDTO requestDTO);
}
